test/Sema/access-level-import-diag-priority.swift

```diff
@@ -0,0 +1,75 @@
+/// Report the most restricted types when many types are problematic.
+
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+
+/// Build the libraries.
+// RUN: %target-swift-frontend -emit-module %t/PublicLib.swift -o %t
+// RUN: %target-swift-frontend -emit-module %t/PackageLib.swift -o %t
+// RUN: %target-swift-frontend -emit-module %t/InternalLib.swift -o %t
+// RUN: %target-swift-frontend -emit-module %t/FileprivateLib.swift -o %t
+// RUN: %target-swift-frontend -emit-module %t/PrivateLib.swift -o %t
+
+/// Check diagnostics.
+// RUN: %target-swift-frontend -typecheck %t/Client.swift -I %t \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify
+// RUN: %target-swift-frontend -typecheck %t/LocalVsImportClient.swift -I %t \
+// RUN:   -enable-experimental-feature AccessLevelOnImport -verify
+
+//--- PublicLib.swift
+public struct PublicImportType {}
+
+//--- PackageLib.swift
+public struct PackageImportType {}
+
+//--- InternalLib.swift
+public struct InternalImportType {}
+
+//--- FileprivateLib.swift
+public struct FileprivateImportType {}
+
+//--- PrivateLib.swift
+public struct PrivateImportType {}
+
+//--- Client.swift
+public import PublicLib
+package import PackageLib // expected-note {{module 'PackageLib' imported as 'package' here}}
+internal import InternalLib // expected-note {{module 'InternalLib' imported as 'internal' here}}
+fileprivate import FileprivateLib // expected-note 2 {{module 'FileprivateLib' imported as 'fileprivate' here}}
+private import PrivateLib // expected-note 2 {{module 'PrivateLib' imported as 'private' here}}
+
+/// Simple ordering
+public func publicFuncUsesPrivate(_ a: PublicImportType, b: PackageImportType, c: InternalImportType, d: FileprivateImportType, e: PrivateImportType) { // expected-error {{function cannot be declared public because its parameter uses a private type}}
```

Comment:
Does this diagnostic indicate which type is the problematic type? I ask because I'm not sure if this only matches a fragment of the diagnostic.

Reply:
That's the whole text of the error, the problematic type is highlighted. I tested the highlights manually as I don't think we have a way to represent them in the verify mode yet, but that would be useful.

Reply:
Got it, I suppose a highlight makes it clear enough.

Reply:
Actually, after that error there is also a note pointing to the type definition. From the command line it would tell the type name clearly, from the IDE probably not that much but the highlight does a decent job there.

And then there's the note on the import if it lowered the access level of the problematic decl.